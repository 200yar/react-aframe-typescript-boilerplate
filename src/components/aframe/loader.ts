export const loaderComponentConfig =  {

  schema: {'default': ''},

  /**
   * Do not use arrow function here
   * "this: any" is needed for the typescript compiler
   * https://github.com/Microsoft/TypeScript/issues/16016#issuecomment-303462193
   */
  init: function(this: any) {
    this.el.addEventListener('loaded', () => {
      console.log('scene loaded');
      const loader = document.getElementById('loader') as HTMLDivElement;
      (loader.parentElement as HTMLDivElement).removeChild(loader);
    })
  }
}
