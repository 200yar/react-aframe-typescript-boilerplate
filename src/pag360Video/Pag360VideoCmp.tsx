//todo: crear componente react para incrustar videos 360 grados. Estaria basado en Aframe o en threejs
//todo: open issue en aframe reepo: isPlaying = true inicialmente
import React from 'react';
import Loader from "../components/loader/LoaderCmp";
import TopMenu from "../components/topMenu/TopMenuCmp";
import SideMenu from "../components/sideMenu/SideMenuCmp";
import {SIDE_MENU_ITEMS} from "../components/sideMenu/SideMenuItems";
import './pag360VideoCmp.css';


export default class Pag360VideoCmp extends React.Component<{}, {isPlaying: boolean}> {

  // todo: "document.querySelector('a-videosphere').isPlaying" seems not to work ("isPlaying" always true)
  public refs: {
    videoEntity: AFrame.Entity,
    sideMenu: SideMenu,
    assets: AFrame.Entity,
    loader: Loader
  };

  public componentDidMount() {
    this.refs.assets && this.refs.assets.addEventListener('loaded', () => {
      this.refs.loader.hide();
    })
  }

  private playVideo = () => {
    this.refs.videoEntity.play();
  }

  private pauseVideo = () => {
    this.refs.videoEntity.pause();
  }

  private openSideMenu = () => {
    this.refs.sideMenu.show();
  }

  public render() {
    return(
      <div>

        <Loader ref="loader">Loading</Loader>

        <SideMenu ref="sideMenu" title="React + AFrame" items={ SIDE_MENU_ITEMS } itemActive="2"/>

        <TopMenu onClickMenuBtn={ this.openSideMenu }>
          <a onClick={ this.playVideo } className="top-menu-item player-btn">Play</a>
          <a onClick={ this.pauseVideo } className="top-menu-item player-btn">Pause</a>
        </TopMenu>

        <a-scene>
          <a-assets ref="assets">
            <video id="videoEntity" ref="videoEntity" src="video/360-fractal-4.mp4" preload="auto"/>
          </a-assets>
          <a-camera reverse-mouse-drag="true" />
          <a-videosphere src="#videoEntity" rotation="0 -90 0"/>
        </a-scene>

      </div>
    )
  }
}