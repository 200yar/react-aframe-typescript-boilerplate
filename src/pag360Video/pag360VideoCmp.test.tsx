import 'aframe';
import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Pag360VideoCmp} from "./Pag360VideoCmp";


describe('Pag360VideoCmp Component', () => {

  const component = shallow(<Pag360VideoCmp>Dummy content</Pag360VideoCmp>);

  it("is defined", () => {
    expect(component).toBeDefined();
  });

  it('renders without crashing', () => {
    expect(component).toBeInstanceOf(ShallowWrapper);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

});