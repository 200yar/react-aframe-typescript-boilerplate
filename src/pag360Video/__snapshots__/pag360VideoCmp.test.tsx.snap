// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pag360VideoCmp Component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pag360VideoCmp>
    Dummy content
</Pag360VideoCmp>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loader>
          Loading
</Loader>,
        <SideMenu
          itemActive="2"
          items={
                    Array [
                              Object {
                                "active": false,
                                "ico": "start.svg",
                                "name": "Home",
                                "url": "#/",
                              },
                              Object {
                                "active": false,
                                "ico": "video-player.svg",
                                "name": "2D/3D Video",
                                "url": "#/2dvideo",
                              },
                              Object {
                                "active": false,
                                "ico": "360-degrees.svg",
                                "name": "360 Video",
                                "url": "#/360video",
                              },
                              Object {
                                "active": false,
                                "ico": "3d-printer.svg",
                                "name": "3D Model Animation",
                                "url": "#/3dmodel",
                              },
                            ]
          }
          title="React + AFrame"
/>,
        <TopMenu
          onClickMenuBtn={[Function]}
>
          <a
                    className="top-menu-item player-btn"
                    onClick={[Function]}
          >
                    Play
          </a>
          <a
                    className="top-menu-item player-btn"
                    onClick={[Function]}
          >
                    Pause
          </a>
</TopMenu>,
        <a-scene>
          <a-assets>
                    <video
                              id="videoEntity"
                              preload="auto"
                              src="video/360-fractal-4.mp4"
                    />
          </a-assets>
          <a-camera
                    reverse-mouse-drag="true"
          />
          <a-videosphere
                    rotation="0 -90 0"
                    src="#videoEntity"
          />
</a-scene>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Loading",
        },
        "ref": "loader",
        "rendered": "Loading",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "itemActive": "2",
          "items": Array [
            Object {
              "active": false,
              "ico": "start.svg",
              "name": "Home",
              "url": "#/",
            },
            Object {
              "active": false,
              "ico": "video-player.svg",
              "name": "2D/3D Video",
              "url": "#/2dvideo",
            },
            Object {
              "active": false,
              "ico": "360-degrees.svg",
              "name": "360 Video",
              "url": "#/360video",
            },
            Object {
              "active": false,
              "ico": "3d-printer.svg",
              "name": "3D Model Animation",
              "url": "#/3dmodel",
            },
          ],
          "title": "React + AFrame",
        },
        "ref": "sideMenu",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <a
              className="top-menu-item player-btn"
              onClick={[Function]}
>
              Play
</a>,
            <a
              className="top-menu-item player-btn"
              onClick={[Function]}
>
              Pause
</a>,
          ],
          "onClickMenuBtn": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Play",
              "className": "top-menu-item player-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Play",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pause",
              "className": "top-menu-item player-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Pause",
            "type": "a",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a-assets>
              <video
                            id="videoEntity"
                            preload="auto"
                            src="video/360-fractal-4.mp4"
              />
</a-assets>,
            <a-camera
              reverse-mouse-drag="true"
/>,
            <a-videosphere
              rotation="0 -90 0"
              src="#videoEntity"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <video
                id="videoEntity"
                preload="auto"
                src="video/360-fractal-4.mp4"
/>,
            },
            "ref": "assets",
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "videoEntity",
                "preload": "auto",
                "src": "video/360-fractal-4.mp4",
              },
              "ref": "videoEntity",
              "rendered": null,
              "type": "video",
            },
            "type": "a-assets",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "reverse-mouse-drag": "true",
            },
            "ref": null,
            "rendered": null,
            "type": "a-camera",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "rotation": "0 -90 0",
              "src": "#videoEntity",
            },
            "ref": null,
            "rendered": null,
            "type": "a-videosphere",
          },
        ],
        "type": "a-scene",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loader>
            Loading
</Loader>,
          <SideMenu
            itemActive="2"
            items={
                        Array [
                                    Object {
                                      "active": false,
                                      "ico": "start.svg",
                                      "name": "Home",
                                      "url": "#/",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "video-player.svg",
                                      "name": "2D/3D Video",
                                      "url": "#/2dvideo",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "360-degrees.svg",
                                      "name": "360 Video",
                                      "url": "#/360video",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "3d-printer.svg",
                                      "name": "3D Model Animation",
                                      "url": "#/3dmodel",
                                    },
                                  ]
            }
            title="React + AFrame"
/>,
          <TopMenu
            onClickMenuBtn={[Function]}
>
            <a
                        className="top-menu-item player-btn"
                        onClick={[Function]}
            >
                        Play
            </a>
            <a
                        className="top-menu-item player-btn"
                        onClick={[Function]}
            >
                        Pause
            </a>
</TopMenu>,
          <a-scene>
            <a-assets>
                        <video
                                    id="videoEntity"
                                    preload="auto"
                                    src="video/360-fractal-4.mp4"
                        />
            </a-assets>
            <a-camera
                        reverse-mouse-drag="true"
            />
            <a-videosphere
                        rotation="0 -90 0"
                        src="#videoEntity"
            />
</a-scene>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Loading",
          },
          "ref": "loader",
          "rendered": "Loading",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "itemActive": "2",
            "items": Array [
              Object {
                "active": false,
                "ico": "start.svg",
                "name": "Home",
                "url": "#/",
              },
              Object {
                "active": false,
                "ico": "video-player.svg",
                "name": "2D/3D Video",
                "url": "#/2dvideo",
              },
              Object {
                "active": false,
                "ico": "360-degrees.svg",
                "name": "360 Video",
                "url": "#/360video",
              },
              Object {
                "active": false,
                "ico": "3d-printer.svg",
                "name": "3D Model Animation",
                "url": "#/3dmodel",
              },
            ],
            "title": "React + AFrame",
          },
          "ref": "sideMenu",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <a
                className="top-menu-item player-btn"
                onClick={[Function]}
>
                Play
</a>,
              <a
                className="top-menu-item player-btn"
                onClick={[Function]}
>
                Pause
</a>,
            ],
            "onClickMenuBtn": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Play",
                "className": "top-menu-item player-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Play",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pause",
                "className": "top-menu-item player-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Pause",
              "type": "a",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a-assets>
                <video
                                id="videoEntity"
                                preload="auto"
                                src="video/360-fractal-4.mp4"
                />
</a-assets>,
              <a-camera
                reverse-mouse-drag="true"
/>,
              <a-videosphere
                rotation="0 -90 0"
                src="#videoEntity"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <video
                  id="videoEntity"
                  preload="auto"
                  src="video/360-fractal-4.mp4"
/>,
              },
              "ref": "assets",
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "videoEntity",
                  "preload": "auto",
                  "src": "video/360-fractal-4.mp4",
                },
                "ref": "videoEntity",
                "rendered": null,
                "type": "video",
              },
              "type": "a-assets",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "reverse-mouse-drag": "true",
              },
              "ref": null,
              "rendered": null,
              "type": "a-camera",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "rotation": "0 -90 0",
                "src": "#videoEntity",
              },
              "ref": null,
              "rendered": null,
              "type": "a-videosphere",
            },
          ],
          "type": "a-scene",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
